.github/workflows/codeql.yml

```diff
@@ -0,0 +1,97 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "develop", "5.x", "6.x", "7.*.x" ]
```

Comment:
The branches `5.x` and `6.x` are no longer maintained, so I'm not sure they should be included here. In a similar way, only the latest branch of `7.*.x` is maintained, i.e. `7.11.x`,  so it should be enough `develop` and `7.11.x`.